ui/components/A11yDialog.vue

```diff
@@ -1,71 +1,94 @@
 <template>
   <div
     :aria-hidden="shown ? `false` : `true`"
-    class="dialog-container"
+    class="fullscreen"
     tabindex="-1"
     @click="overlayClick">
-    <div class="dialog-overlay" tabindex="-1" />
-
-    <dialog
-      :id="`${id}-dialog`"
-      :aria-labelledby="id + '-dialog-title'"
-      :open="shown"
-      class="card">
-      <div>
-
-        <a
-          v-if="cancellable"
-          href="#close"
-          class="icon-button close"
-          title="Close"
-          @click.prevent="hide">
-          Close
-          <OflSvg name="close" />
-        </a>
-
-        <h2 :id="`${id}-dialog-title`" tabindex="-1" autofocus>{{ title }}</h2>
-
-        <slot />
-
-      </div>
-    </dialog>
+    <div class="backdrop" tabindex="-1" />
+
+    <div class="dialog-wrapper" :class="{ wide }">
+      <dialog
+        :id="`${id}-dialog`"
+        :aria-labelledby="id + '-dialog-title'"
+        :open="shown"
+        class="card">
+        <div>
+
+          <a
+            v-if="cancellable"
+            href="#close"
+            class="icon-button close"
+            title="Close"
+            @click.prevent="hide">
+            Close
+            <OflSvg name="close" />
+          </a>
+
+          <h2 :id="`${id}-dialog-title`" tabindex="-1" autofocus>
+            {{ title }}
+            <slot name="after-title" />
+          </h2>
+
+          <slot />
+
+        </div>
+      </dialog>
+    </div>
   </div>
 </template>
 
 <style lang="scss" scoped>
-.dialog-container {
+.fullscreen {
   outline: none;
-}
-
-.dialog-overlay {
-  z-index: 1000;
-  background-color: rgba(0, 0, 0, 0.66);
   position: fixed;
+  z-index: 1000;
   top: 0;
   left: 0;
-  bottom: 0;
   right: 0;
+  bottom: 0;
+  /* fixes bug with hiding app bars on mobile */
+  /* (avoid 'width: 100vw' as this includes the scroll bar) */
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.backdrop {
+  background-color: rgba(0, 0, 0, 0.66);
+  position: absolute;
+  width: 100%;
+  height: 100%;
 }
 
-.dialog-container[aria-hidden=true],
-[data-a11y-dialog-native] > .dialog-overlay {
+.fullscreen[aria-hidden=true],
+[data-a11y-dialog-native] > .backdrop {
   display: none;
 }
 
+.fullscreen > .dialog-wrapper {
```

Comment:
Do you really need to add a wrapper div? Adding the padding and `display: flex` to `.fullscreen` could also work.